src/SharpCompress/Common/Zip/ZipFilePart.cs

```diff
@@ -126,16 +126,11 @@ protected Stream CreateDecompressionStream(Stream stream)
 
         protected Stream GetCryptoStream(Stream plainStream)
         {
-            if ((Header.CompressedSize == 0)
-#if !NO_CRYPTO
- && ((Header.PkwareTraditionalEncryptionData != null)
-                    || (Header.WinzipAesEncryptionData != null)))
-#else
-                && (Header.PkwareTraditionalEncryptionData != null))
-#endif
+            if ((Header.CompressedSize == 0) && !string.IsNullOrEmpty(Header.Password))
```

Comment:
This doesn't make sense.  The lack of a password should be it's own `if` statement.  Not `AND`ed with this check.